ingestion/portland_ingest.py

```diff
@@ -21,131 +21,164 @@
 
-calendar_df["day_type"] = calendar_df["date"].dt.day_name()
-
-day_type_mapping = {"Saturday": "Saturday", "Sunday": "Sunday"}
-
-calendar_df["day_type"] = (
-    calendar_df["day_type"].map(day_type_mapping).fillna("Weekday")
-)
+def load_ridership_datasets(file_names):
+    """
+    Takes a dictionary in the form key: season_day type
+    and value: file name, loads the corresponding ridership
+    data and creates a column with the corresponding
+    season value and day type (Saturday, Weekday, or Sunday),
+    and returns a dictionary with a key (season_day) and value
+    (the corresponding dataframe).
+    """
+    data_frames = {}
+    for key, value in file_names.items():
+        df = load_data(folder_path + value)
+        # get season from filename
+        if "spring" in value:
+            df["season"] = "Spring"
+        elif "summer" in value:
+            df["season"] = "Summer"
+        elif "fall" in value:
+            df["season"] = "Fall"
+        elif "winter" in value:
+            df["season"] = "Winter"
+
+        # get day_type from filename
+        if "sat" in value:
+            df["day_type"] = "Saturday"
+        elif "weekday" in value:
+            df["day_type"] = "Weekday"
+        elif "sunday" in value:
+            df["day_type"] = "Sunday"
+        data_frames[key] = df
+    return data_frames
+
+
+def format_data(all_rider_data):
+    """
+    standarize column formats and keep the relevant
+    data fields needed to match our data model, or
+    merge the data at the datetime level
+    """
+    for key, df in all_rider_data.items():
+        df.columns = df.columns.str.lower().str.replace(" ", "_")
+        all_rider_data[key] = df[["location_id", "ons", "day_type", "season"]]
+    return all_rider_data
 
-# flag seasons, all days are flagged to Spring, except for summer months,
-# to differenciate for routes that only run during the summer months
-calendar_df["season"] = "Spring"
 
-# Summer 2023 date range
-summer_start = pd.to_datetime("2023-06-21")
-summer_end = pd.to_datetime("2023-09-22")
+def create_calendar_df(start_date, end_date, season_ranges):
+    """
+    Create a dataframe with datetime values as well as
+    the corresponding season and day type associated with
+    each datetime, return a dataframe
+    """
+    calendar_dates = pd.date_range(start=start_date, end=end_date, freq="D")
+    calendar_df = pd.DataFrame(calendar_dates, columns=["date"])
+    calendar_df["day_type"] = calendar_df["date"].dt.day_name()
+    calendar_df["day_type"] = (
+        calendar_df["day_type"]
+        .map({"Saturday": "Saturday", "Sunday": "Sunday"})
+        .fillna("Weekday")
+    )
+    calendar_df["season"] = "Spring"
+
+    # label seasons based on date ranges
+    for season, (season_start, season_end) in season_ranges.items():
+        calendar_df.loc[
+            (calendar_df["date"] >= season_start) & (calendar_df["date"] < season_end),
```

Comment:
I would double check that season_start and season_end are correct.
If you create a datetime.datetime object for 2023.02.21 for example it creates an object representing 2023-02-21 00:00:00:00 so its important to check that it matches and I'm not sure it does with the current seasons dictionary.